File: runtime/ruby/units/wasi/write_string_list.rb
# requires: memory/iws, memory/iwsb, memory/init
def write_string_list(strings, list_ptr, buf_ptr)
  strings.each_with_index do |s, i|
    @memory.iws(list_ptr + i * 4, buf_ptr)
    @memory.init(buf_ptr, s, 0, s.bytesize)
    @memory.iwsb(buf_ptr + s.bytesize, 0)
    buf_ptr += s.bytesize + 1
  end
  ERRNO_SUCCESS
end

File: crates/dewasm-backend-python/units/wasi/write_string_list.py
# requires: memory/iws, memory/iwsb, memory/init
def write_string_list(self, strings, list_ptr, buf_ptr):
    for i, s in enumerate(strings):
        self.memory.iws(list_ptr + i * 4, buf_ptr)
        self.memory.init(buf_ptr, s, 0, len(s))
        self.memory.iwsb(buf_ptr + len(s), 0)
        buf_ptr += len(s) + 1
    return self.ERRNO_SUCCESS
